File: 通讯协议/ProtoGen/output/handlers/csharp_all/ActivityRevelryHandler.cs
/*-----------------------------------------------
 *本文件由代码生成器自动生成，
 *千万不要修改本文件的任何代码，
 *修改的的任何代码都会被覆盖掉！
 --------------------------------------------------*/

using System.IO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Pomelo.DotNetClient;

namespace pomelo.revelry
{
	public class ActivityRevelryHandler
	{
		private PomeloClient _socket;

		public RevelryGetColumnResponse lastRevelryGetColumnResponse { get { return _socket.GetLastResponse<RevelryGetColumnResponse>(); } }
		public RevelryGetRankInfoResponse lastRevelryGetRankInfoResponse { get { return _socket.GetLastResponse<RevelryGetRankInfoResponse>(); } }
		public RevelryExchangeResponse lastRevelryExchangeResponse { get { return _socket.GetLastResponse<RevelryExchangeResponse>(); } }
		static ActivityRevelryHandler()
		{
			EventTypes.RegistRequestType("revelry.activityRevelryHandler.revelryGetColumnRequest", typeof(RevelryGetColumnRequest), typeof(RevelryGetColumnResponse));
			EventTypes.RegistRequestType("revelry.activityRevelryHandler.revelryGetRankInfoRequest", typeof(RevelryGetRankInfoRequest), typeof(RevelryGetRankInfoResponse));
			EventTypes.RegistRequestType("revelry.activityRevelryHandler.revelryExchangeRequest", typeof(RevelryExchangeRequest), typeof(RevelryExchangeResponse));
		}
		public ActivityRevelryHandler(PomeloClient socket)
		{
			this._socket = socket;

		}
		public void revelryGetColumnRequest(Action<PomeloException,RevelryGetColumnResponse> cb,object option = null)
		{
			var request = new RevelryGetColumnRequest();
			_socket.request<RevelryGetColumnResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as RevelryGetColumnResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void revelryGetRankInfoRequest(string id,Action<PomeloException,RevelryGetRankInfoResponse> cb,object option = null)
		{
			var request = new RevelryGetRankInfoRequest();
			request.id= id;
			_socket.request<RevelryGetRankInfoResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as RevelryGetRankInfoResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}
		public void revelryExchangeRequest(int id,int num,Action<PomeloException,RevelryExchangeResponse> cb,object option = null)
		{
			var request = new RevelryExchangeRequest();
			request.id= id;
			request.num= num;
			_socket.request<RevelryExchangeResponse>(request, (object msg, out int s2c_code, out string s2c_msg) => {
				var rsp = msg as RevelryExchangeResponse;
				s2c_code = rsp.s2c_code;
				s2c_msg = rsp.s2c_msg;
				return s2c_code == 200;
			}, cb, option);

		}

	}


}

File: 通讯协议/ProtoGen/output/handlers/lua/ActivityRevelryHandler.lua
-------------------------------------------------
-- 本文件由代码生成器自动生成，
-- 千万不要修改本文件的任何代码，
-- 修改的的任何代码都会被覆盖掉！
----------------------------------------------------

local Socket = require "Xmds.Pomelo.LuaGameSocket"
require "base64"
require "activityRevelryHandler_pb"

--------------------------------Pomelo-------------------------------
Pomelo = Pomelo or {}

--------------------------------ActivityRevelryHandler-------------------------------
Pomelo.ActivityRevelryHandler = {}

local function revelryGetColumnRequestEncoder(msg)
	local input = activityRevelryHandler_pb.RevelryGetColumnRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function revelryGetColumnRequestDecoder(stream)
	local res = activityRevelryHandler_pb.RevelryGetColumnResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityRevelryHandler.revelryGetColumnRequest(cb,option)
	local input = nil
	Socket.OnRequestStart("revelry.activityRevelryHandler.revelryGetColumnRequest", option)
	Socket.Request("revelry.activityRevelryHandler.revelryGetColumnRequest", input, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityRevelryHandler.lastRevelryGetColumnResponse = res
			Socket.OnRequestEnd("revelry.activityRevelryHandler.revelryGetColumnRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] revelry.activityRevelryHandler.revelryGetColumnRequest decode error!!"
			end
			Socket.OnRequestEnd("revelry.activityRevelryHandler.revelryGetColumnRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, revelryGetColumnRequestEncoder, revelryGetColumnRequestDecoder)
end


local function revelryGetRankInfoRequestEncoder(msg)
	local input = activityRevelryHandler_pb.RevelryGetRankInfoRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function revelryGetRankInfoRequestDecoder(stream)
	local res = activityRevelryHandler_pb.RevelryGetRankInfoResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityRevelryHandler.revelryGetRankInfoRequest(id,cb,option)
	local msg = {}
	msg.id = id
	Socket.OnRequestStart("revelry.activityRevelryHandler.revelryGetRankInfoRequest", option)
	Socket.Request("revelry.activityRevelryHandler.revelryGetRankInfoRequest", msg, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityRevelryHandler.lastRevelryGetRankInfoResponse = res
			Socket.OnRequestEnd("revelry.activityRevelryHandler.revelryGetRankInfoRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] revelry.activityRevelryHandler.revelryGetRankInfoRequest decode error!!"
			end
			Socket.OnRequestEnd("revelry.activityRevelryHandler.revelryGetRankInfoRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, revelryGetRankInfoRequestEncoder, revelryGetRankInfoRequestDecoder)
end


local function revelryExchangeRequestEncoder(msg)
	local input = activityRevelryHandler_pb.RevelryExchangeRequest()
	protobuf.FromMessage(input,msg)
	return (input:SerializeToString())
end

local function revelryExchangeRequestDecoder(stream)
	local res = activityRevelryHandler_pb.RevelryExchangeResponse()
	res:ParseFromString(stream)
	return res
end

function Pomelo.ActivityRevelryHandler.revelryExchangeRequest(id,num,cb,option)
	local msg = {}
	msg.id = id
	msg.num = num
	Socket.OnRequestStart("revelry.activityRevelryHandler.revelryExchangeRequest", option)
	Socket.Request("revelry.activityRevelryHandler.revelryExchangeRequest", msg, function(res)
		if(res.s2c_code == 200) then
			Pomelo.ActivityRevelryHandler.lastRevelryExchangeResponse = res
			Socket.OnRequestEnd("revelry.activityRevelryHandler.revelryExchangeRequest", true)
			cb(nil,res)
		else
			local ex = {}
			if(res.s2c_code) then
				ex.Code = res.s2c_code
				ex.Message = res.s2c_msg
			else
				ex.Code = 501
				ex.Message = "[LuaXmdsNetClient] revelry.activityRevelryHandler.revelryExchangeRequest decode error!!"
			end
			Socket.OnRequestEnd("revelry.activityRevelryHandler.revelryExchangeRequest", false,ex.Code,ex.Message)
			cb(ex,nil)
		end
	end, revelryExchangeRequestEncoder, revelryExchangeRequestDecoder)
end




--------------------------------Pomelo-------------------------------
return Pomelo
